packages/server/database/migrations/20210118184615-removeDuplicateOrgUsers.ts

```diff
@@ -0,0 +1,51 @@
+import {R} from 'rethinkdb-ts'
+export const up = async function (r: R) {
+  const now = new Date()
+  try {
+    const duplicateUserIds = await r
+      .table('OrganizationUser')
+      .filter({removedAt: null})
+      .group('userId', 'orgId')
+      .count()
+      .ungroup()
+      .filter((row) => row('reduction').gt(1))('group')
+      .map((row) => row.nth(0)) // just return the userIds, not the orgIds
+      .distinct()
+      .run()
+
+    const duplicateOrgUsers = await r
+      .table('OrganizationUser')
+      .getAll(r.args(duplicateUserIds), {index: 'userId'})
+      .filter({removedAt: null})
+      .group(
+        'userId',
+        'orgId'
+      )('id')
+      .run()
+
+    const duplicatesToRemove = duplicateOrgUsers
+      .map((user) => user['reduction'].slice(1, user['reduction'].length))
+      .filter((user) => user.length)
+      .flat()
+
+    await r
```

Comment:
how do we account for billing? i assume some of these duplicates are pro/enterprise?

Reply:
When removing an org user, I'm adding a `removedReason` field. Once the migration has been shipped, I was thinking I could do a query to check which users have the `removedReason` 'duplicate' . I imagine the list will be small and I could share that list with the sales team who can organise a refund.

Reply:
that's good for things in the past, but we'll still need to tell stripe that their seat count has decreased, otherwise stripe will keep charging the same price

Reply:
Update pushed with Redis lock and updating Stripe! Two questions:

1. Do we only need to update Stripe and not the `Invoice` or `InvoiceItemHook` db tables? It looks like the `Invoice` fields are generated from info from Stripe [here](https://github.com/ParabolInc/parabol/blob/fix-replicated-user/packages/server/billing/helpers/generateInvoice.ts#L394) but I wanted to check

2. Can I access our Stripe account? I haven't been able to test whether the Stripe has actually updated following the migration